bridge/irc_manager.go

```diff
@@ -253,7 +255,7 @@ func (m *IRCManager) generateNickname(discord DiscordUser) string {
 	suffix := m.bridge.Config.Suffix
 	newNick := nick + suffix
 
-	useFallback := len(newNick) > ircnick.MAXLENGTH || m.bridge.ircListener.DoesUserExist(newNick)
+	useFallback := len(newNick) > m.nickLength || m.bridge.ircListener.DoesUserExist(newNick)
```

Comment:
I know what you have now is better practice (and less racey), but I think it would be cleaner to access it directly from the struct:

```suggestion
	useFallback := len(newNick) > b.bridge.Config.MaxNickLength || m.bridge.ircListener.DoesUserExist(newNick)
```

This has the immediate benefit of supporting hot config reload for this setting